flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamTwoInputProcessor.java

```diff
@@ -293,9 +291,12 @@ private void updateAvailability(InputStatus status, StreamTaskInput input) {
 	}
 
 	private void checkAndSetAvailable(int inputIndex) {
-		StreamTaskInput input = getInput(inputIndex);
+		CompletableFuture<?> inputAvailable = getInput(inputIndex).isAvailable();
```

Comment:
Since this is performance-critical, I'd avoid a call to isAvailable if `status == InputStatus.END_OF_INPUT`.

I'd split the check into two and have a short-cut check on EOF.

Reply:
It might be a good idea. I will try it out.